encoding/mkv.md

```diff
@@ -0,0 +1,505 @@
+(as long as the track order is the same).
+
+*For the audiophiles who like to keep their anime music lossless,
+we've encoded only the OP/EDs with FLAC.
+This leaves the episode to be compressed more efficiently with a lossy codec
+such as opus, for those who don't like 'wasting' space with lossless audio.
+Since we are encoding around 90% <!-- TODO: recalculate --> of the show with opus,
+and only have the OP/ED + NCOP/NCED files occurring once,
+the filesize difference and lossless music should satisfy both groups.
+Normal dialogue should be nearly transparent at 128 kbps with opus,
+but this comes with some caveats.
+If there are 'insert' songs within an episode that you'd like to keep lossless,
+you will need to encode the entire episode with a lossless codec.*
+
+
+## Working with ordered chapters
+
+*At this point, you should have encoded clips of each episode
+plus the OP/ED/NCOP/NCED encodes.
+The following assumes these clips are named in the same `ep##.264` format
+as the .m2ts files.
+In our working directory, we should also have subtitle tracks, `ep##.ass`,
+and the encoded audio tracks `ep##_cut.opus`.
+If you used the above audio-cutting commands,
+chapter files in plaintext should also be in the folder.*
+
+At this point, you can delete the .wav files and move the .m2ts files
+out into their own folder.
+Your working directory should now look like this:
+
+![Clean Working Directory](images/mkv/example_directory.png)
+
+
+### Muxing
+
+Before we can start using ordered chapters,
+we will have to mux the current files into mkv's.
+This is because ordered chapters reference other mkv's by their UID
+to generate the virtual timeline for the player.
+MKVToolNix can save chapters *into* existing mkv files,
+so this step will be the **only time** we mux.
+
+This can be done with the MKVToolNix GUI,
+in which case the order of the tracks in the bottom list is important.
+Although mpv will only warn that tracks are mismatched,
+it is a good idea to keep the same track order in all of the mkv's.
+In our example, we will use
+*Video*, *Audio*, then *Subtitles*
+as the order for every file.
+
+The order can be changed with the arrows or by dragging the tracks
+in the bottom box of the *Mulitplexer* window in the GUI:
+
+![Track order matters!](images/mkv/arrows_to_change_track_order.png)
+
+This can also be done automatically using the command line.
+An example Python script (with no attachments):
+
+```py
+import shlex, subprocess
+
+video_eps = ['ep{:02d}'.format(i) for i in range(1, 14)]
+video_op = ['ed', 'nced', 'ncop', 'op']
+videos = video_eps + video_op
+
+audio_eps = ['ep{:02d}_cut.opus'.format(i) for i in range(1, 14)]
+audio_op = ['ed_cut.flac', 'nced_cut.flac', 'ncop_cut.flac', 'op_cut.flac']
+audios = audio_eps + audio_op
+
+for n, a in zip(videos, audios):
+    args = shlex.split('mkvmerge -o {0}.mkv {0}.264 {1} {0}.ass'.format(n, a))
+    subprocess.run(args)
```

Comment:
Same as discussed in discord. :wink: Build a list and run that. Also, the internal variable name for the first parameter is `cmd` and that better fits what we're building here.

Reply:
It seems to be 
```py
def run(args: Union[bytes, str, Sequence[Union[bytes, str, PathLike]]], ...
```
so maybe what you're thinking of is some other library or an older version of subprocess?

Reply:
Maybe I was confusing things, then. I still find "args" to be a weird choice because it obviously contains the executable as the first element, but my argument regarding the internal name seems to be debunked.